src/opml.cpp

```diff
@@ -178,29 +178,31 @@ void rec_find_rss_outlines(
 	}
 }
 
-bool opml::import(
+nonstd::optional<std::string> opml::import(
 	const std::string& filename,
 	FileUrlReader& urlcfg)
 {
 	xmlDoc* doc = xmlReadFile(filename.c_str(), nullptr, 0);
 	if (doc == nullptr) {
-		return false;
+		return strprintf::fmt(_("Error: Failed to parse OPML file \"%s\""), filename);
 	}
 
-	xmlNode* root = xmlDocGetRootElement(doc);
+	nonstd::optional<std::string> error_message;
 
+	xmlNode* root = xmlDocGetRootElement(doc);
 	for (xmlNode* node = root->children; node != nullptr;
 		node = node->next) {
 		if (strcmp((const char*)node->name, "body") == 0) {
 			LOG(Level::DEBUG, "opml::import: found body");
 			rec_find_rss_outlines(urlcfg, node->children, "");
-			urlcfg.write_config();
+			error_message =
+				urlcfg.write_config(); // Only an error if optional string has a value
```

Comment:
Shouldn't this exit the loop if an error occurred?

Also, the comment confused me at first because I thought it explains something specific to just this particular line of code. I think this comment can be removed after the docs are updated, because the docs will explain the same thing as this comment.

Reply:
That indeed makes sense, thanks.